[Expo 42] Bare app build failing

### Summary

Android app fails to build on `Task :unimodules-react-native-adapter:compileDebugJavaWithJavac`

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

_No response_

### Environment

  Expo CLI 4.7.2 environment info:
    System:
      OS: Windows 10 10.0.19043
    Binaries:
      Node: 14.17.1 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
    SDKs:
      Android SDK:
        API Levels: 26, 29, 30
        Build Tools: 29.0.2, 30.0.3
        System Images: android-28 | Google Play Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-S | Google Play Intel x86 Atom_64
    IDEs:
      Android Studio: Version  4.2.0.0 AI-202.7660.26.42.7351085
    npmPackages:
      expo: ~42.0.0 => 42.0.0
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: ~0.63.4 => 0.63.4
      react-native-web: ~0.13.12 => 0.13.18
    Expo Workflow: bare

### Reproducible demo or steps to reproduce from a blank project

Initialize a new Expo Bare App
Setup typescript (only creating the tsconfig.json and installing the deps when prompted by `expo start`)
Add eslint (config universe/native and the plugin simple-import-sort)
Try to `yarn android` and it shoud fail on the task mentioned above

Sorry for the delay, I'm getting the exact same error as you, could the reason be that I've installed node via choco?

```
require(expo-modules-autolinking)(process.argv.slice(1))
        ^

ReferenceError: expo is not defined
    at [eval]:1:9
    at Script.runInThisContext (vm.js:134:12)
    at Object.runInThisContext (vm.js:311:38)
    at internal/process/execution.js:77:19
    at [eval]-wrapper:6:22
    at evalScript (internal/process/execution.js:76:60)
    at internal/main/eval_string.js:23:3
C:\Users\ivopr\test\node_modules\@unimodules\react-native-adapter\android\src\main\java\org\unimodules\adapters\react\ModuleRegistryAdapter.java:26: error: cannot find symbol
    mModuleRegistryProvider = new ReactModuleRegistryProvider(new ExpoModulesPackageList().getPackageList(), null);
                                                                  ^
  symbol:   class ExpoModulesPackageList
  location: class ModuleRegistryAdapter
```

I'll try changing node

--- a/packages/@unimodules/react-native-adapter/CHANGELOG.md
+++ b/packages/@unimodules/react-native-adapter/CHANGELOG.md
@@ -10,6 +10,8 @@
 
 ### 🐛 Bug fixes
 
+- Escape Android autolinking script for Windows. ([#13494](https://github.com/expo/expo/pull/13494) by [@byCedric](https://github.com/byCedric))
+
 ### 💡 Others
 
 ## 6.3.0 — 2021-05-25

--- a/packages/@unimodules/react-native-adapter/scripts/autolinking.gradle
+++ b/packages/@unimodules/react-native-adapter/scripts/autolinking.gradle
@@ -94,7 +94,7 @@ class ExpoAutolinkingManager {
     String[] args = [
       'node',
       '--eval',
-      'require("expo-modules-autolinking")(process.argv.slice(1))',
+      'require(\'expo-modules-autolinking\')(process.argv.slice(1))',
       '--',
       command,
       '--platform',

